### ISSUE TITLE
\lnot ¬ should be a prefix operator

### ISSUE THREAD

**Issue by rzach** [brucemiller/LaTeXML#2163]
`$\lnot A$` will generate `<mi mathvariant="normal">¬</mi><mo>⁢</mo><mi>A</mi>` but the ¬ is a prefix operator not an identifier.

In general, is there a way to help LaTeXML generate the right MathML for prefix operators? E.g., if I want to take a relation (say, `\sim`) and turn it into a prefix operator (i.e., avoid the spacing around relations in LaTeX) I'd say `\mathord{sim}` but that turns it into an `<mi>`. My screen reader, e.g., will read out `$\mathord{sim}A$ as "tilde times upper A".

**Comment by dginev** [comment]
Interesting, it is defined as a `FUNCTION` role at the moment, as is `\neg`:

https://github.com/brucemiller/LaTeXML/blob/2dde370c97d94238e2f987e0ef5fe50acc67cb58/lib/LaTeXML/Package/TeX.pool.ltxml#L5912

We have a `POSTFIX` role for constructs such as the `!` factorial, so maybe we need a very simple `PREFIX` role alongside for simple prefix symbols meant to become `<mo>` elements.

**Comment by dginev** [comment]
@rzach you can indicate a prefix operator using `\mathop` at the moment, although it uses the `BIGOP` role internally which assumes a bit too much (or at least the name is too suggestive for larger operators such as `\forall`, `\exists`).

Here's a quick test:

```
$ latexmlc --whatsin=math 'literal:\mathop{\lnot} a'
$ latexmlc --pmml --whatsin=math 'literal:\mathop{\lnot} a'
```

**Comment by rzach** [comment]
Thanks! (While we're on the subject of ¬ : Why does it have the `mathvariant="normal"` attribute? (It makes screen readers read it as "normal not sign" instead of "not sign" which gets tedious after a while--at least when run through MathJax.)

**Comment by dginev** [comment]
@rzach the mathvariant details are neither here nor there, since the emitted markup is incorrect, as we discussed above.

In general, the "normal" variant is now the only accepted mathvariant value in [MathML Core](https://www.w3.org/TR/mathml-core/#the-mathvariant-attribute), and is used to avoid the default "italic" treatment of `<mi>` elements. I believe latexml explicitly sets that for content which it isn't convinced should be italic.

So indeed, we don't need that attribute, but the root of the cause is that we are treating an operator the same way we would a function name.

**Comment by brucemiller** [comment]
Not convinced that the original markup is *wrong*, but could be better.  Probably should have declared `\not` and `\lnot` as `role='OPERATOR'`, rather than `FUNCTION`.   I think that will lead to an `mo` rather than `mi`, and so the `mathvariant` would no longer be needed, even though that also is not wrong.  Pronouncing it seems wrong though.  Using `\mathord` gives you an "ordinary" symbol, so is not the way to mark up a prefix operator.

**Comment by rzach** [comment]
I see there are commands in https://github.com/brucemiller/LaTeXML/blob/master/lib/LaTeXML/texmf/latexml.sty that look like they should allow me to (re)define \lnot (and other symbols) in a way that tells LaTeXML what MathML stuff they should map to. But they're not documented there or in the manual afaict...

**Comment by brucemiller** [comment]
Not quite prepared to dive into documentation, ATM, but perhaps a fix will help? #2170;  Thanks for the report!

**Comment by rzach** [comment]
Aside: BIGOP is better since it turns ¬ into an `<mo>`. BUT LaTeXML adds an explicit ` rspace="0.167em"`.

LaTeXML also has the PREFIX role (according to https://math.nist.gov/~BMiller/LaTeXML/manual/math/details/roles.html) which afaict isn't actually used anywhere. I figured out that I can get a plain `<mo>` without the extra space using `\lxMathTweak{role={POSTFIX}}{\lnot}`. (It did give me a 
```
Warning:not_parsed:>POSTFIX MathParser failed to match rule 'Anything'...
```
Perhaps the clean way to do this is to actually let LaTeXML do something with the PREFIX role (e.g., turn it into `<mo form="prefix">`) and then make `\lnot` have that role. That's probably what @dginev meant up top in the first comment...

Anyway, thanks!
